core/src/avm1/return_value.rs

```diff
@@ -93,7 +95,18 @@ impl<'gc> ReturnValue<'gc> {
         match self {
             Immediate(val) => Ok(val),
             ResultOf(frame) => {
-                avm.run_current_frame(context, frame)?;
+                match avm.run_current_frame(context, frame) {
+                    Err(ExecutionError::ScriptError(e)) => {
+                        return Err(e);
+                    }
+                    Err(e) => {
+                        log::warn!(
+                            "Couldn't resolve value, encountered an avm1 execution error: {}",
+                            e
+                        );
+                    }
+                    _ => {}
+                }
 
                 Ok(avm.pop())
```

Comment:
Is this right? Shouldn't this bail out in case of other errors as well (say, if we got `InvalidSwf`, there's not any recovery from that, right?). Particularly if we hit an error, I doubt a return value was pushed on the stack, so falling through to `avm.pop()` seems troublesome.

Reply:
I'm not sure. My view on this PR is that I think a bunch of these behaviours are wrong, but I'm certain that they're, in general at least, less wrong than before.

Looking at the code, I can see that only the following errors can theoretically raise here:
- `ScriptError` of course, no issues
- `FrameNotOnStack`
- `AlreadyExecutingFrame`
- `NoStackFrame`

The last 3 do feel more like an issue in our code more than with a particular SWF. Long term I'd love to find a way to make these not possible by API design. For now, I'd be okay with making this use the same error we'd use for user-initiated script aborts - bubble up a `Error::Aborted` and halt further script execution.

Does that sound okay?

Reply:
Just noticed that `InvalidSwf` isn't actually raised anywhere anymore, think it was refactored away. Removing it with a commit. That leaves `ExecutionError` to just the errors listed above and I still consider them mostly to be programmer errors.

Reply:
Bubbling up `Error::Aborted` sounds fine to me.

Reply:
Decided to merge the two errors, it felt weird having a generic "something happened!" and we'd otherwise have two Errors that nest eachother. Also implemented the halting logic, and amended a test to prove it.